spec/index.bs

```diff
@@ -4095,6 +4118,123 @@ configuring each of the [=render stages=].
+If a primitive is clipped, however, the output values assigned to vertices produced by clipping are clipped.
+
+Considering an edge between vertices |a| and |b| that got clipped, resulting in the vertex |c|,
+let's define |t| to be the ratio between the edge vertices:
+|c|.p = |t| &times; |a|.p &plus; (1 &minus; |t|) &times; |b|.p,
+where |x|.p is the output [=clip position=] of a vertex |x|.
+
+For each vertex output value "v" with a corresponding fragment input,
+|a|.v and |b|.v would be the outputs for |a| and |b| vertices respectively.
+The clipped shader output |c|.v is produced based on the interpolation qualifier:
+<dl class="switch">
+    : "flat"
+    ::
+        Flat interpolation is unaffected, and is based on <dfn dfn>provoking vertex</dfn>,
+        which is the first vertex in the primitive. The output value is the same
+        for the whole primitive, and matches the vertex output of the [=provoking vertex=]:
+        |c|.v = [=provoking vertex=].v
+
+    : "linear"
+    ::
+        The interpolation ratio gets adjusted against the perspective coordinates of the
+        [=clip position=]s, so that the result of interpolation is linear in screen space:
+
+        |c|.v = |t| &times; |a|.v &times; |c|.p.w &divide; |a|.p.w &plus; (1 &minus; |t|) &times; |b|.v &times; |c|.p.w &divide; |b|.p.w
+
+        Here, |x|.p.w is the "w" component of the [=clip position=] of a vertex |x|.
+
+    : "perspective"
+    ::
+        The value is linearly interpolated in clip space, producing perspective-correct values:
+
+        |c|.v = |t| &times; |a|.v &plus; (1 &minus; |t|) &times; |b|.v
+</dl>
+
+Issue: link to interpolation qualifiers in WGSL
+
+### Rasterization ### {#rasterization}
+
+Rasterization is the hardware processing stage that maps the generated primitives
+to the 2-dimensional rendering area.
+
+1. First, the clipped vertices are transformed into <dfn dfn>NDC</dfn> - normalized device coordinates.
+    Given the output position |p|, the [=NDC=] coordinates are computed as:
+
+    ndc(|p|) = vector(|p|.x &divide; |p|.w, |p|.y &divide; |p|.w, |p|.z &divide; |p|.w)
+
+2. Let |renderExtent| be [=RenderPassDescriptor/renderExtent=] of the active render pass.
+
+    Then the [=NDC=] coordinates |n| are converted into framebuffer coordinates, based on the size of the render targets:
+
+    framebufferCoords(n) = 0.5&times;vector((|n|.x&plus;1) &times; |renderExtent|.{{GPUExtent3DDict/width}}, (|n|.y&plus;1) &times; |renderExtent|.{{GPUExtent3DDict/height}})
+
+
+#### Polygon Rasterization #### {#polygon-rasterization}
+
+Let |v|(|i|) be the framebuffer coordinates for the clipped vertex number |i| (starting with 1)
+in a rasterized polygon of |n| vertices.
+
+Note: this section uses the term "polygon" instead of a "triangle",
+since {#primitive-clipping} stage may have introduced additional vertices.
+This is non-observable by the application.
```

Comment:
SInce it is non-observable maybe we could remove it and do the clipping as part of the rasterization? Also what happens when the fragment writes gl_FragDepth?

Reply:
Only the fact that we are dealing with a "polygon" is non-observable. The clipping stage itself is still necessary before we can talk about rasterization.

Reply:
> Also what happens when the fragment writes gl_FragDepth?

So far, we haven't reached the FS, so for this stage it doesn't matter if the user writes anything to the frag depth.